src/services/workflow.service.js

```diff
@@ -292,18 +317,15 @@ class WorkflowService {
     // Reset.
     const initialQuery = print(baseSubscriber.query.query)
     subscription.query.query = baseSubscriber.query.query
-    subscription.actionNames = baseSubscriber.query.actionNames
-    subscription.tearDownActionNames = baseSubscriber.query.tearDownActionNames
+    subscription.callbacks = baseSubscriber.query.callbacks
 
-    subscribers.slice(1)
-      .forEach(subscriber => {
-        if (!isEqual(subscriber.query.variables, baseSubscriber.query.variables)) {
-          throw new Error('Error recomputing subscription: Query variables do not match.')
-        }
-        this.mergeQueries(baseSubscriber.query.query, subscriber.query.query)
-        subscription.actionNames = union(subscription.actionNames, subscriber.query.actionNames)
-        subscription.tearDownActionNames = union(subscription.tearDownActionNames, subscriber.query.tearDownActionNames)
-      })
+    for (const subscriber of subscribers.slice(1)) {
+      if (!isEqual(subscriber.query.variables, baseSubscriber.query.variables)) {
+        throw new Error('Error recomputing subscription: Query variables do not match.')
+      }
+      this.mergeQueries(baseSubscriber.query.query, subscriber.query.query)
+      subscription.callbacks = union(subscription.callbacks, subscriber.query.callbacks)
```

Comment:
Note to self: manual test, and confirm this is working OK.

Reply:
Ha! I knew it. Quickly wrote a unit test to verify it.

![image](https://user-images.githubusercontent.com/304786/132263739-35bbfab1-cc44-4ff0-bab4-4fc5aeeee39a.png)

Let me fix it, and make the test permanent.